pkg/compressor/compressor.go

```diff
@@ -55,11 +55,12 @@ func CompressSnapshot(data io.ReadCloser, compressionPolicy string) (io.ReadClos
 		defer pWriter.CloseWithError(err)
 		defer gWriter.Close()
 		defer data.Close()
-		_, err = io.Copy(gWriter, data)
+		n, err := io.Copy(gWriter, data)
```

Comment:
Actually error `err` must be declared before the `pWriter.CloseWithError(err)` . so when error will occur in `io.Copy()` we have to capture that error and pass it to the pWriter.CloseWithError(err)`.
https://golang.org/pkg/io/#PipeWriter.CloseWithError

Reply:
Yes. Updated the PR with the necessary changes